### ISSUE TITLE
iOS: Preference for marking story as read only after time or manually

### ISSUE THREAD

**Issue by whispy** [samuelclay/NewsBlur#1596]
Issue: When I'm reading a long item (in either Story or Text modes) over several days (which means I'm going in and out of the app many times), Newsblur marks the item as read before I've completely read the item. This occurs because Newsblur has no sense of how far I've scrolled within an item. (At least, I believe it doesn't — sometimes it saves my position when reopening an item, but other times it starts me from the beginning. I haven't been able to figure out why this behavior changes.)

Request: Add an option to track scroll percentage within an item when in Story or Text mode, and only mark item as read when passing that percentage threshold.

I'm happy to provide more info about this idea.

**Comment by samuelclay** [comment]
I think it's hard to know, as a user, how you'd want to customize a percentage read. It should probably just be a mark-read-when-scrolled-to-bottom of story. 

On the web, we have a manual mode so no story is marked as read until you do by tapping on it or hitting 'u'. 

Would you want a setting to turn off marking as read when reading a story? You'd have to swipe it to mark as read, or hit the menu when reading the story, or use a keyboard shortcut (on ipad). 

Alternatively, we can add a button to mark it as read if you turn on the don't mark as read feature. Would that work, maybe a read status toggle at top and bottom?

**Comment by whispy** [comment]
> I think it's hard to know, as a user, how you'd want to customize a percentage read. It should probably just be a mark-read-when-scrolled-to-bottom of story.
> 
> On the web, we have a manual mode so no story is marked as read until you do by tapping on it or hitting 'u'.
> 
> Would you want a setting to turn off marking as read when reading a story? You'd have to swipe it to mark as read, or hit the menu when reading the story, or use a keyboard shortcut (on ipad).
> 
> Alternatively, we can add a button to mark it as read if you turn on the don't mark as read feature. Would that work, maybe a read status toggle at top and bottom?

Thanks for the thoughts, Samuel! I agree with your point that determining and setting a percentage threshold would be more trouble than it's worth for a user.

Your suggestion to completely turn off mark-as-read when reading a story would be a good solution, especially when paired with a big `Mark As Read` button at the end of each feed item. Nice idea!

**Comment by samuelclay** [comment]
Let's build this on iOS first:

* New preference (matches the web): 

<img width="668" alt="Screen Shot 2022-01-25 at 11 48 06 AM" src="https://user-images.githubusercontent.com/44229/151021367-de548a49-3352-44af-8c81-3317e1e1a556.png">

* Manual means a new "Mark story read" button, basically the same button as the Share/Save/Train buttons, but full width, above those three. And this button can be built into the story, so add it to the html and handle clicks through JS
* Hitting that button changes the text to "Read" and a slightly dimmer text color.
* Tapping it again marks unread, and restores text to "Mark story read"

And we should also do the timer, so if you remain on a story for more than X seconds, it auto-marks as read. Nifty!

**Comment by whispy** [comment]
Woohoo, thanks! Only thing is I use Android, so I won't be able to beta test this for you. Otherwise, sounds great!

**Comment by samuelclay** [comment]
I'll duplicate this for Android...

**Comment by Dejal** [comment]
Done.

![Screenshot 2024-07-30 at 15 02 59](https://github.com/user-attachments/assets/54b3997a-40da-4989-90ff-db061ca491a3)
![Screenshot 2024-07-30 at 15 03 31](https://github.com/user-attachments/assets/16417c44-53f2-4c01-84db-a59b5c3a6a0b)

**Comment by samuelclay** [comment]
Ok, can I release `dejal` or `dejal-alpha` on iOS? Do you think it's ready, and if I do so, are you around to handle anything that comes up?

**Comment by samuelclay** [comment]
This is great, but the default should be on scroll.

**Comment by Dejal** [comment]
You can always release `dejal` anytime. The `dejal-alpha` changes should be tested, but may be fine to release if all looks good; you can merge into `dejal` and `master`. But you mentioned a few things for #1847 (in `dejal-alpha`), so I'd suggest only releasing `dejal` for now. It'd be great to get the Mac app released.

I'll be traveling for the next few days, but will be available from Monday 5th.

**Comment by Dejal** [comment]
I just checked, and the default is `scroll`. If you're seeing something else that would likely be a setting you changed.

![Screenshot 2024-08-05 at 12 37 02](https://github.com/user-attachments/assets/97dd3e55-2afe-4f92-9e50-86525b03470c)

I installed on a new Simulator device and confirmed that `scroll` was used.
